packages/cubejs-clickhouse-driver/driver/ClickHouseDriver.js

```diff
@@ -90,10 +90,42 @@ class ClickHouseDriver extends BaseDriver {
     
     return this.withConnection((connection, queryId) => {
       return connection.querying(formattedQuery, { dataObjects: true, queryOptions: { query_id: queryId, join_use_nulls: 1 } })
-        .then(res => res.data);
+        .then(res => this._normaliseResponse(res));
     });
   }
 
+  _normaliseResponse(res) {
+    //
+    //
+    //  ClickHouse returns DateTime as strings in format "YYYY-DD-MM HH:MM:SS" in UTC timezone
+    //  cube.js expects them in format "YYYY-DD-MMTHH:MM:SS.000Z", so translate them based on the metadata returned
+    //
+    //  ClickHouse returns some number types as js numbers, others as js string, normalise them all to strings  
+    //
+    //
+    if (res.data) {
+      res.data.forEach(row=>{
+        for (let field in row) {
+          let value = row[field]
+          if (value !== null) {
+            let meta = res.meta.find(m=>m.name == field)
+            if (meta.type.includes("DateTime")) {
+              row[field] = value.replace(" ", "T") + ".000Z"
+            } 
+            else if (meta.type.includes("Date")) {
+              row[field] = value + "T00:00:00.000Z"
+            } 
+            else if (meta.type.includes("Int") || meta.type.includes("Float")) {
+              // convert all numbers into strings
+              row[field] = `${value}`
```

Comment:
Why numbers have to be converted to strings?

Reply:
lots of the cubejs-schema-compiler tests that deal with measures assert that they are strings.  

examples 

- https://github.com/statsbotco/cube.js/blob/73db73f874582ad86b3b01d4e93a04c3385e7003/packages/cubejs-schema-compiler/test/DataSchemaCompilerTest.js#L158
- https://github.com/statsbotco/cube.js/blob/73db73f874582ad86b3b01d4e93a04c3385e7003/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L318
- https://github.com/statsbotco/cube.js/blob/73db73f874582ad86b3b01d4e93a04c3385e7003/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L365
- https://github.com/statsbotco/cube.js/blob/73db73f874582ad86b3b01d4e93a04c3385e7003/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L400
- https://github.com/statsbotco/cube.js/blob/73db73f874582ad86b3b01d4e93a04c3385e7003/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L534

Reply:
I believe that this is due to the fact that the range of values that the javascript type supports can overflow.  Clickhouse caters for this by returning strings for UInt64,Int64 already. I assume postgres does the same, and assumed that this was the protocol for cube.js

Reply:
As I understand this is only an issue with `Int64` and `UInt64` data types. How is it handled for Postgres which also has [support](https://www.postgresql.org/docs/11/datatype-numeric.html) for 64-bit numbers?

Reply:
@paveltiunov what should we be doing here with numbers vs strings ?

Reply:
From what I understand the Postgres driver returns strings for 64-bit integers and actual numbers for other types. This seems to be consistent with default ClickHouse behaviour.

Reply:
We probably have some variations from driver to driver but as a rule of thumb all numeric types should be returned as strings due to JS can't handle correctly big integers and decimal numbers at all.

Also regarding Timestamp format: driver should always return only local dates in `YYYY-DD-MMTHH:MM:SS.sss` format as all conversions are done in SQL. There's only one case driver can return UTC: it's UTC timestamp representation in cube.js server local timezone. Otherwise there should be no 'Z' at the end.

Reply:
Seems like something to be added to driver development guidelines :)

Reply:
@inikolaev Yep. Definitely worth adding it.

Reply:
Re timestamps - I am confused as for example

The following test uses timezone America/Los_Angeles 

https://github.com/statsbotco/cube.js/blob/bde5ea1dc493f4af599c6fcdcaecbcd399dfcebd/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L302

However the timestamp asserted is UTC (including a Z)

https://github.com/statsbotco/cube.js/blob/bde5ea1dc493f4af599c6fcdcaecbcd399dfcebd/packages/cubejs-schema-compiler/test/GraphBuilderTest.js#L315

Actually - Every date in that test is formatted the same.

Is there an example

Reply:
@cameronbraid Yep. It's confusing. The thing is Postgres is that exception from the rule: it provides UTC timestamp representation in cube.js server local timezone. Actually it causing problems so we're considering to fix this in the Postgres driver itself.